scripts/deploy/update_version_numbers.rb

```diff
@@ -35,13 +35,17 @@ def update_version()
 end
 
 def update_changelog()
-    date = Time.now.strftime("%Y-%m-%d")
+    changelog_contents = File.read("CHANGELOG.md")
 
-    replace_in_file(
-        "CHANGELOG.md",
-        /## XX.XX.XX - 20XX-XX-XX/,
-        "## XX.XX.XX - 20XX-XX-XX\n\n## #{@version} - #{date}"
-    )
+    if (!changelog_contents.include? @version)
```

Comment:
Could this fail if we are releasing `20.50.10` and `20.50.1` already exists in the file?

Reply:
yep, but in practice we've never had a patch number .10, so I think this is ok (we can optimize later if we need to)